core/templates/pages/community-dashboard-page/services/contribution-opportunities.service.ts

```diff
@@ -17,73 +17,108 @@
+import { Injectable } from '@angular/core';
 
-angular.module('oppia').factory('ContributionOpportunitiesService', [
-  'ContributionOpportunitiesBackendApiService',
-  function(ContributionOpportunitiesBackendApiService) {
-    var skillOpportunitiesCursor = null;
-    var translationOpportunitiesCursor = null;
-    var voiceoverOpportunitiesCursor = null;
-    var moreSkillOpportunitiesAvailable = true;
-    var moreTranslationOpportunitiesAvailable = true;
-    var moreVoiceoverOpportunitiesAvailable = true;
+import { ContributionOpportunitiesBackendApiService,
+  IFetchOpportunitySuccessCallbackParams }
+// eslint-disable-next-line max-len
+  from 'pages/community-dashboard-page/services/contribution-opportunities-backend-api.service';
+import { SkillOpportunity } from
+  'domain/opportunity/SkillOpportunityObjectFactory';
 
-    var _getSkillOpportunities = function(cursor, successCallback) {
-      ContributionOpportunitiesBackendApiService.fetchSkillOpportunities(
-        cursor).then((opportunities, nextCursor, more) => {
-        skillOpportunitiesCursor = nextCursor;
-        moreSkillOpportunitiesAvailable = more;
-        successCallback(opportunities, more);
-      });
-    };
-    var _getTranslationOpportunities = function(
-        languageCode, cursor, successCallback) {
-      ContributionOpportunitiesBackendApiService.fetchTranslationOpportunities(
-        languageCode, cursor).then((opportunities, nextCursor, more) => {
-        translationOpportunitiesCursor = nextCursor;
-        moreTranslationOpportunitiesAvailable = more;
-        successCallback(opportunities, more);
+@Injectable({
+  providedIn: 'root'
+})
+export class ContributionOpportunitiesService {
+  skillOpportunitiesCursor = null;
+  translationOpportunitiesCursor = null;
+  voiceoverOpportunitiesCursor = null;
+  moreSkillOpportunitiesAvailable = true;
+  moreTranslationOpportunitiesAvailable = true;
+  moreVoiceoverOpportunitiesAvailable = true;
+
+  constructor(
+    private contributionOpportunitiesBackendApiService:
+      ContributionOpportunitiesBackendApiService) { }
+
+  private _getSkillOpportunities(cursor: string,
+      successCallback: (opportunities: SkillOpportunity[],
+      more: boolean) => void): void {
+    this.contributionOpportunitiesBackendApiService.fetchSkillOpportunities(
+      cursor).then((value: IFetchOpportunitySuccessCallbackParams) => {
+      this.skillOpportunitiesCursor = value.nextCursor;
+      this.moreSkillOpportunitiesAvailable = value.more;
+      successCallback(value.opportunities, value.more);
+    });
+  }
+
+  private _getTranslationOpportunities(languageCode: string, cursor: string,
+      successCallback: (opportunities: SkillOpportunity[],
+      more: boolean) => void): void {
+    this.contributionOpportunitiesBackendApiService
+      .fetchTranslationOpportunities(languageCode, cursor)
+      .then((value: IFetchOpportunitySuccessCallbackParams) => {
```

Comment:
IFecth**ed**Opportunit**ies**SuccessCallbackParams

Reply:
@h-ARTS, is this comment resolved?

Reply:
@DubeySandeep This has been changed. PTAL file `contribution-opportunities.service.ts`

Reply:
@DubeySandeep Since you told me to make separate interfaces now this one is resolved too
**PTAL**